nnunetv2/imageio/natural_image_reader_writer.py

```diff
@@ -56,13 +56,13 @@ def read_images(self, image_fnames: Union[List[str], Tuple[str, ...]]) -> Tuple[
             print('Image files:')
             print(image_fnames)
             raise RuntimeError()
-        return np.vstack(images).astype(np.float32), {'spacing': (999, 1, 1)}
+        return np.vstack(images, dtype=np.float32, casting='unsafe'), {'spacing': (999, 1, 1)}
```

Comment:
Can you comment on the use of 'unsafe', any specific reason this is picked?

Reply:
I've chosen 'unsafe' for backwards compatibility as `astype` uses 'unsafe' casting (https://numpy.org/doc/stable/reference/generated/numpy.ndarray.astype.html). 
Moreover, 'unsafe' casting is needed in this case because numpy raises an exception when casting from int32 to float32:
```py
>>> import numpy as np
>>> x = np.array([1, 2, 3], dtype=np.int32)
>>> x.astype(np.float32)
array([1., 2., 3.], dtype=float32)
>>> x.astype(np.float32, casting='safe') 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Cannot cast array data from dtype('int32') to dtype('float32') according to the rule 'safe'
```